Mathlib/Topology/Order/ScottTopology.lean

```diff
@@ -0,0 +1,366 @@
+
+variable {α β}
+
+variable [Preorder α] [Preorder β]
+
+/--
+A set `u` is said to be inaccessible by directed joins if, when the least upper bound of a directed
+set `d` lies in `u` then `d` has non-empty intersection with `u`.
+-/
+def DirSupInacc (u : Set α) : Prop :=
+  ∀ ⦃d : Set α⦄, d.Nonempty → DirectedOn (· ≤ ·) d → ∀ ⦃a : α⦄, IsLUB d a → a ∈ u → (d ∩ u).Nonempty
+
+namespace Topology
+/--
+The Scott-Hausdorff topology is defined as the topological space where a set `u` is open if, when
+the least upper bound of a directed set `d` lies in `u` then there is a tail of `d` which is a
+subset of `u`.
+-/
+def scottHausdorff : TopologicalSpace α where
+  IsOpen u := ∀ ⦃d : Set α⦄, d.Nonempty → DirectedOn (· ≤ ·) d → ∀ ⦃a : α⦄, IsLUB d a →
+    a ∈ u → ∃ b ∈ d, Ici b ∩ d ⊆ u
+  isOpen_univ := fun d ⟨b, hb⟩ _ _ _ _ => ⟨b, hb, (Ici b ∩ d).subset_univ⟩
+  isOpen_inter s t hs ht d hd₁ hd₂ a hd₃ ha := by
+    obtain ⟨b₁, hb₁d, hb₁ds⟩ := hs hd₁ hd₂ hd₃ ha.1
+    obtain ⟨b₂, hb₂d, hb₂dt⟩ := ht hd₁ hd₂ hd₃ ha.2
+    obtain ⟨c, hcd, hc⟩ := hd₂ b₁ hb₁d b₂ hb₂d
+    refine ⟨c, hcd, ?_⟩
+    · calc
+        Ici c ∩ d ⊆ Ici b₁ ∩ Ici b₂ ∩ d := by
+        { apply inter_subset_inter_left d
+          apply subset_inter (Ici_subset_Ici.mpr hc.1) (Ici_subset_Ici.mpr hc.2) }
+        _ = (Ici b₁ ∩ d) ∩ (Ici b₂ ∩ d) := by rw [inter_inter_distrib_right]
+        _ ⊆ s ∩ t := inter_subset_inter hb₁ds hb₂dt
+  isOpen_sUnion := fun s h d hd₁ hd₂ a hd₃ ⟨s₀, hs₀s, has₀⟩ => by
+    obtain ⟨b, hbd, hbds₀⟩ := h s₀ hs₀s hd₁ hd₂ hd₃ has₀
+    exact ⟨b, hbd, Set.subset_sUnion_of_subset s s₀ hbds₀ hs₀s⟩
+
+lemma ScottHausdorff.isOpen_of_isLowerSet {s : Set α} (h : IsLowerSet s) :
+    scottHausdorff.IsOpen s := fun d ⟨b, hb⟩ _ _ hda ha =>
+      ⟨b, hb,Subset.trans (inter_subset_right (Ici b) d)
+      (fun _ hc => h (mem_upperBounds.mp hda.1 _ hc) ha)⟩
+
+/--
+The Scott topology is defined as the join of the topology of upper sets and the Scott Hausdorff
+topology.
+-/
+def scott : TopologicalSpace α := upperSet α ⊔ scottHausdorff
+
+lemma upperSet_le_scott : upperSet α ≤ scott := le_sup_left
+
+lemma scottHausdorff_le_scott : @scottHausdorff α ≤ @scott α := le_sup_right
+
+end Topology
+
+end preorder
+
+namespace Topology
+/--
+Type synonym for a preorder equipped with the Scott topology
+-/
+def WithScott := α
+
+variable {α β}
+
+namespace WithScott
+
+/-- `toScott` is the identity function to the `WithScott` of a type.  -/
+@[match_pattern] def toScott : α ≃ WithScott α := Equiv.refl _
+
+/-- `ofScott` is the identity function from the `WithScott` of a type.  -/
+@[match_pattern] def ofScott : WithScott α ≃ α := Equiv.refl _
+
+@[simp] lemma toScott_symm_eq : (@toScott α).symm = ofScott := rfl
+@[simp] lemma ofScott_symm_eq : (@ofScott α).symm = toScott := rfl
+@[simp] lemma toScott_ofScott (a : WithScott α) : toScott (ofScott a) = a := rfl
+@[simp] lemma ofScott_toScott (a : α) : ofScott (toScott a) = a := rfl
+-- porting note: removed @[simp] to make linter happy
```

Comment:
I doubt there's any point putting porting notes in code that was never merged to mathlib in Lean 3.

Also in this case you should tag this `@[simp, nolint simpNF]` since the linter is wrong: your lemma is useful for `dsimp` and can't be proved by `dsimp`.

Reply:
Done, thanks.